src/cmd/policy.go

```diff
@@ -21,6 +24,63 @@ type regoInput struct {
 	Files []string `json:"files"`
 }
 
+type gitlabRepoMetadata struct {
+	Name        string             `json:"name,omitempty"`
+	Description string             `json:"description,omitempty"`
+	Language    string             `json:"language,omitempty"`
+	Languages   map[string]float64 `json:"languages,omitempty"`
+}
+
+type githubRepoMetadata struct {
+	Name        string             `json:"name,omitempty"`
+	Description string             `json:"description,omitempty"`
+	Language    string             `json:"language,omitempty"`
+	Languages   map[string]float64 `json:"languages,omitempty"`
+}
+
+type genericStruct interface {
```

Comment:
i would give this a better name - maybe something related to AST marshaling - maybe `astMarshallable` or something